service/history/decisionTaskHandler.go

```diff
@@ -219,9 +220,17 @@ func (handler *decisionTaskHandlerImpl) handleDecisionScheduleActivity(
 		return err
 	}
 
-	_, _, err = handler.mutableState.AddActivityTaskScheduledEvent(handler.decisionTaskCompletedID, attr)
+	dispatchLocally := handler.config.EnableRequestActivityLocalDispatchByDomain(handler.domainEntry.GetInfo().Name) &&
```

Comment:
This logic should be put inside AddActivityTaskScheduledEvent

Reply:
You have TODO below to return additional information anyways. So you can use that to decide if you need to ad the started event.

Reply:
agree, lets to update decision handler func's to return interface{} to collect all that info